presto-main/src/main/java/com/facebook/presto/metadata/FunctionManager.java

```diff
@@ -200,6 +200,17 @@ public void registerBuiltInFunctions(List<? extends BuiltInFunction> functions)
                 .collect(toImmutableList());
     }
 
+    public Collection<? extends SqlFunction> getFunctions(TransactionId transactionId, QualifiedFunctionName functionName)
```

Comment:
The implementation here also suggested that `TransactionId` does not need to be required. `FunctionNamespaceManager` would happily take an empty transaction handle. So why do you enforce transaction ID as input? It happened to always exists should not be the reason you always require it to exist.